docs/AA--Javascript-in-EE/components/form/select/model-select.md

```diff
@@ -0,0 +1,186 @@
+            // by th react select component (see props section)
+            name: uniqueId( 'model-select-' ),
+            isClearable: true,
+            isLoading: true,
+            placeholder: 'Select...'
+        }
+    }
+    modelEntities={ [] }
+    modelName={ '' }
+    mapOptionsCallback={ buildOptions } // see props section for details
+    optionsEntityMap={ null }
+    queryData={ { limit: 100, order: 'desc' } }
+    selectLabel={ '' }
+    addAllOptionLabel={ '' }
+    getQueryString={ () => '' }
+/>
+```
+
+## `ModelSelect` Props
+
+### selectConfiguration
+
+This is an object that can contain any values components higher in the tree want to pass down to the `react-select` component.  You can read about all the possible values that can be passed down via [this document](https://deploy-preview-2289--react-select.netlify.com/props).
+
+The following options in this object are generated and set in the ModelSelect component:
+
+| Prop | Purpose |
+|------|---------|
+`name` | This is automatically set to be a unique name for the select component. It can be overridden.
+`options` | This is derived from the incoming array of EE model entity objects.
+`value` | This is derived from the incoming `defaultValue` value on `selectConfiguration`.
+
+
+- Type: `Object`
+- Required: No
+- Default:
+```js
+{
+    name: uniqueId( 'model-select-' ),
+    isClearable: true,
+    isLoading: true,
+    placeholder: 'Select...'
+}
+```
+
+### modelEntities
+
+This is expected to be an array of entities for the model corresponding to the value in the `modelName` prop.  Typically this is simply the response from the REST request for that model.
+
+- Type: `Array`
+- Required: No
+- Default: `[]`
+
+### modelName
+
+This informs what model the incoming modelEntities are for.  This should correspond to the canonical modelName for models as respresented in the EE Rest schema (i.e. Events are `event`, Datetimes are `datetime`).
+
+- Type: `String`
+- Required: No
+- Default: `''`
+
+### mapOptionsCallback
+
+This should be a function that converts incoming modelEntities into an array of option objects usable by the react-select component.  By default, this is the built-in [`buildOptions`](../../../../../assets/src/components/form/select/build-options.js) callback and that can be used as a reference for how any custom function should behave.
+
+- Type: `Function`
+- Required: No
+- Default: `buildOptions`
+
+### optionsEntityMap:
+
+If null, this is ignored and the default map found in [build-options.js](../../../../../src/components/form/select/build-options.js) is used.  When provided, it is expected to be an object that helps inform how the `mapOptionsCallback` should map entity objects to the `value`, and `label` pairs in the final array of option objects. The map should be shaped like this:
+
+```js
+const map = {
+    event: {
+        label: 'EVT_name',
+        value: 'EVT_ID',
+    }
+};
```

Comment:
I wonder if this should be abstracted more first and then followed by an example:

```jsx
const map = {
    modelName: {
        label: 'model-field-to-use-for-label',
        value: 'model-field-to-use-for-value',
    }
};
// example
const map = {
    event: {
        label: 'EVT_name',
        value: 'EVT_ID',
    }
};
```